image-hd.c

```diff
@@ -166,6 +199,160 @@ static int hdimage_insert_ebr(struct image *image, struct partition *part)
 	return 0;
 }
 
+static const char *
+gpt_partition_type_lookup(const char *str)
```

Comment:
The string-length check is outside of this function, so I think the argument should just be the single `char`.